extensions/schema-compare/src/dialogs/schemaCompareDialog.ts

```diff
@@ -365,36 +501,60 @@ export class SchemaCompareDialog {
+			this.sourceTextBox.value = this.sourceDacpacPath;
 			this.formBuilder.removeFormItem(this.sourceServerComponent);
 			this.formBuilder.removeFormItem(this.sourceDatabaseComponent);
+			this.formBuilder.removeFormItem(this.sourceProjectFilePathComponent);
 			this.formBuilder.insertFormItem(this.sourceDacpacComponent, 2, { horizontal: true, titleFontSize: titleFontSize });
 			this.dialog.okButton.enabled = await this.shouldEnableOkayButton();
 		});
 
 		// show server and db dropdowns
 		this.sourceDatabaseRadioButton.onDidClick(async () => {
-			this.sourceIsDacpac = false;
+			this.sourceEndpointType = mssql.SchemaCompareEndpointType.Database;
 			this.formBuilder.insertFormItem(this.sourceServerComponent, 2, { horizontal: true, titleFontSize: titleFontSize });
 			this.formBuilder.insertFormItem(this.sourceDatabaseComponent, 3, { horizontal: true, titleFontSize: titleFontSize });
 			this.formBuilder.removeFormItem(this.sourceDacpacComponent);
+			this.formBuilder.removeFormItem(this.sourceProjectFilePathComponent);
 
 			await this.populateServerDropdown(false);
 		});
 
+		// show project directory browser
+		this.sourceProjectRadioButton.onDidClick(async () => {
+			this.sourceEndpointType = mssql.SchemaCompareEndpointType.Project;
+			this.sourceTextBox.value = this.sourceProjectFilePath;
+			this.formBuilder.removeFormItem(this.sourceServerComponent);
+			this.formBuilder.removeFormItem(this.sourceDatabaseComponent);
+			this.formBuilder.removeFormItem(this.sourceDacpacComponent);
+			this.formBuilder.insertFormItem(this.sourceProjectFilePathComponent, 2, { horizontal: true, titleFontSize: titleFontSize });
+			this.dialog.okButton.enabled = await this.shouldEnableOkayButton();
+		});
+
 		// if source is currently a db, show it in the server and db dropdowns
 		if (this.schemaCompareMainWindow.sourceEndpointInfo && this.schemaCompareMainWindow.sourceEndpointInfo.endpointType === mssql.SchemaCompareEndpointType.Database) {
 			this.sourceDatabaseRadioButton.checked = true;
-			this.sourceIsDacpac = false;
-		} else {
+			this.sourceEndpointType = mssql.SchemaCompareEndpointType.Database;
+		} else if (this.schemaCompareMainWindow.sourceEndpointInfo && this.schemaCompareMainWindow.sourceEndpointInfo.endpointType === mssql.SchemaCompareEndpointType.Dacpac) {
 			this.sourceDacpacRadioButton.checked = true;
-			this.sourceIsDacpac = true;
+			this.sourceEndpointType = mssql.SchemaCompareEndpointType.Dacpac;
+		} else if (this.schemaCompareMainWindow.sourceEndpointInfo) {
+			this.sourceProjectRadioButton.checked = true;
+			this.sourceEndpointType = mssql.SchemaCompareEndpointType.Project;
 		}
+
 		let flexRadioButtonsModel = this.view.modelBuilder.flexContainer()
 			.withLayout({ flexFlow: 'column' })
-			.withItems([this.sourceDacpacRadioButton, this.sourceDatabaseRadioButton])
+			.withItems([this.sourceDacpacRadioButton, this.sourceProjectRadioButton, this.sourceDatabaseRadioButton])
```

Comment:
@kisantia @dzsquared Any thoughts on ordering of the options?  Could be odd to insert the new one in the middle... but this will probably be the least-used option, and the first/last buttons are easier to access.

Reply:
I'm thinking about the order more as what is the logical order of the relationship:
database = no files
dacpac = 1 file
project = many files

or, to keep the current order:
dacpac = 1 object
database = 1 object
database project = many objects

Since the list is relatively short, I don't think the first/last ease of selection tax is worth much.  I think database project at the bottom is my preferred option.